File: core/src/main/java/javabot/operations/DictOperation.java
package javabot.operations;

import com.antwerkz.maven.SPI;

@SPI(BotOperation.class)
public class DictOperation extends UrlOperation {
    @Override
    protected String getBaseUrl() {
        return "http://dictionary.reference.com/browse/";
    }

    @Override
    protected String getTrigger() {
        return "dict ";
    }
}

File: src/main/kotlin/javabot/operations/DictOperation.kt
package javabot.operations

import jakarta.inject.Inject
import javabot.Javabot
import javabot.dao.AdminDao

class DictOperation @Inject constructor(bot: Javabot, adminDao: AdminDao) :
    UrlOperation(bot, adminDao) {
    override fun getBaseUrl(): String {
        return "http://dictionary.reference.com/browse/"
    }

    override fun getTrigger(): String {
        return "dict "
    }
}
